src/models/relations.js

```diff
@@ -111,6 +121,49 @@ export default class Relations {
         eventsFromSender.push(event);
     }
 
+    /**
+     * For relations that are about to be redacted, remove them from aggregation
+     * data sets and emit an update event.
+     *
+     * @param {MatrixEvent} redactedEvent
+     * The original relation event that is about to be redacted.
+     */
+    _onBeforeRedaction = (redactedEvent) => {
```

Comment:
This doesn't seem to handle the case of a redaction failing to apply? (would expect the count to go down optimistically then back up due to failure to redact)

Reply:
This is runs after the server has approved the redaction, but before we have marked the original event redacted, so we can trust it has happened here. https://github.com/vector-im/riot-web/issues/9675 will cover local echo for redactions, but the current code doesn't need to worry about that right now.

Reply:
a comment to help future explorers might be worthwhile

Reply:
Comment [added](https://github.com/matrix-org/matrix-js-sdk/pull/911/commits/f411d50253ff344080641c2de0a750f389e22190).